launchable/utils/session.py

```diff
@@ -1,102 +1,103 @@
-
-    # Jenkins pipeline also launches every process as a new session, so better to scope this to a particular build.
-    # For freestyle job types that do not do this, session IDs do not distinguish different builds, so that is a problem, too.
-    #
-    # See: https://github.com/jenkinsci/lib-durable-task/blob/6e020747205cb5aca5af757bc0d2a302c42bdb79/src/cmd/bash/durable_task_monitor.go#L41
-    # See: https://www.jenkins.io/doc/book/pipeline/jenkinsfile/#using-environment-variables
-    if os.environ.get("JENKINS_URL") is not None:
-        id = os.environ.get("BUILD_URL")
-        if id is not None:
-            return id
-        Logger().warning("JENKINS_URL environment variable is set but not BUILD_URL")
-
-    if sys.platform == "win32":
-        import wmi  # type: ignore
-        c = wmi.WMI()
-        wql = "Associators of {{Win32_Process='{}'}} Where Resultclass = Win32_LogonSession Assocclass = Win32_SessionProcess".format(
-            os.getpid())
-        res = c.query(wql)
-        return str(res[0].LogonId)
-    else:
-        return str(os.getsid(os.getpid()))
+    return Path(os.environ.get(SESSION_DIR_KEY) or os.getcwd()).expanduser()
+
+
+def _session_file_path() -> Path:
+    return _session_file_dir() / ".launchable"
+
+
+def read_build() -> Optional[str]:
+    f = _session_file_path()
+    try:
+        if not f.exists():
+            return None
+
+        with open(str(_session_file_path())) as session_file:
+            session = json.load(session_file)
+            return session["build"]
+
+    except Exception as e:
+        raise Exception("Can't read {}".format(f)) from e
 
 
 def read_session(build_name: str) -> Optional[str]:
-    f = _session_file_path(build_name)
+    f = _session_file_path()
     try:
         if not f.exists():
             return None
-        return f.read_text()
+
+        with open(str(_session_file_path())) as session_file:
+            session = json.load(session_file)
+            if build_name != session.get('build', None):
+                raise Exception("Build name is different from saved. input. input:{} saved:{}".format(
+                    build_name, session.get('build', None)))
+
+            return session.get("session")
+
     except Exception as e:
         raise Exception("Can't read {}".format(f)) from e
 
 
-def write_session(build_name: str, session_id: str) -> None:
+def write_build(build_name: str) -> None:
     try:
         if not _session_file_dir().exists():
             _session_file_dir().mkdir(parents=True, exist_ok=True)
 
-        _session_file_path(build_name).write_text(session_id)
+        session = {}
+        session["build"] = build_name
+
+        with open(str(_session_file_path()), 'w') as session_file:
+            json.dump(session, session_file)
+
+    except Exception as e:
+        raise Exception("Can't write to {}. Perhaps set the {} environment variable to specify an alternative writable path?".format(
```

Comment:
I might suggest "Can't write to {}. Is the path writable? If not, set the {} environment variable to specify an alternative directory for this file"